src/test/java/com/baidu/hugegraph/unit/rest/RestClientTest.java

```diff
@@ -404,4 +407,72 @@ public void testClose() {
         Assert.assertNotNull(cleanExecutor);
         Assert.assertTrue(cleanExecutor.isShutdown());
     }
+
+    @Test
+    public void testAuthContext() {
+        RestClientImpl client = new RestClientImpl("/test", 1000, 10, 5, 200);
+        Assert.assertNull(client.getAuthContext());
+
+        String token = UUID.randomUUID().toString();
+        client.setAuthContext(token);
+        Assert.assertEquals(token, client.getAuthContext());
+
+        client.resetAuthContext();
+        Assert.assertNull(client.getAuthContext());
+    }
+
+    private static class MockRestClientImpl extends AbstractRestClient {
+
+        public MockRestClientImpl(String url, int timeout) {
+            super(url, timeout);
+        }
+
+        @Override
+        protected void checkStatus(Response response,
+                                   Response.Status... statuses) {
+            // pass
+        }
+    }
+
+    @Test
+    public void testRequest() {
+        RestClient client = new MockRestClientImpl("test", 1000);
+
+        WebTarget target = Mockito.mock(WebTarget.class);
+        Builder builder = Mockito.mock(Builder.class);
+
+        Mockito.when(target.path("test")).thenReturn(target);
+        Mockito.when(target.path("test")
+                           .path(AbstractRestClient.encode("id")))
+               .thenReturn(target);
+        Mockito.when(target.path("test").request()).thenReturn(builder);
+        Mockito.when(target.path("test")
+                           .path(AbstractRestClient.encode("id"))
+                           .request())
+               .thenReturn(builder);
+
+        Response response = Mockito.mock(Response.class);
+        Mockito.when(response.getStatus()).thenReturn(200);
+        Mockito.when(response.getHeaders())
+               .thenReturn(new MultivaluedHashMap<>());
+        Mockito.when(response.readEntity(String.class)).thenReturn("content");
+        Mockito.when(builder.delete()).thenReturn(response);
+        Mockito.when(builder.get()).thenReturn(response);
+
+        Whitebox.setInternalState(client, "target", target);
+
+        RestResult result;
+
+        result = client.delete("test", ImmutableMap.of());
+        Assert.assertEquals(200, result.status());
+
+        result = client.delete("test", "id");
```

Comment:
test for each method:

1. setContext with different token
2. do get/post/put/delete
3. assert AUTHORIZATION header is expected

and also test it after resetAuthContext()